internal/rbd/controllerserver.go

```diff
@@ -1078,7 +1078,9 @@ func (cs *ControllerServer) CreateSnapshot(
 
 	// Fetch source volume information
 	rbdVol, err := GenVolFromVolID(ctx, req.GetSourceVolumeId(), cr, req.GetSecrets())
-	defer rbdVol.Destroy()
+	if rbdVol != nil {
+		defer rbdVol.Destroy()
```

Comment:
this could be moved after `if err != nil`, no need to check if `rbdVol != nil` in that case.